Cauldron/Assets/Scripts/ColoredByBrewedPotion.cs

```diff
@@ -0,0 +1,195 @@
+﻿using System.Collections;
+using UnityEngine;
+using Zenject;
+
+namespace CauldronCodebase
+{
+    public class ColoredByBrewedPotion : MonoBehaviour
+    {
+        [Tooltip("Color after accepted or declined potion")]
+        [SerializeField] private Color neutralColor = Color.gray;
+        
+        [Tooltip("Color for Placebo potion brewed")]
+        [SerializeField] private Color placeboColor = Color.HSVToRGB(0.97f, 0.68f, 0.27f);
+        
+        [Range(-1f, 1f)] [SerializeField] private float saturation = 0f;
+        [Range(-1f, 1f)] [SerializeField] private float brightness = 1f;
+        
+        [Tooltip("Delay in seconds")]
+        [SerializeField] private float delay = 0f; 
+        
+        [SerializeField] private bool tween = false;  
+        
+        [Tooltip("Tween time in seconds")]
+        [SerializeField] private float tweenTime = 1f; 
+
+        private RecipeProvider recipeProvider;
+        private Cauldron cauldron;
+
+        [SerializeField] private SpriteRenderer spriteRenderer;
+        [SerializeField] private ParticleSystem particleSystem;
+
+
+        [Inject]
+        private void Construct(Cauldron cauldron, RecipeProvider recipeProvider)
+        {
+            this.recipeProvider = recipeProvider;
+            this.cauldron = cauldron;
+        }
+
+        private void OnValidate()
+        {
+            spriteRenderer = GetComponent<SpriteRenderer>();
+            particleSystem = GetComponent<ParticleSystem>();
+            if (!tween)
+            {
+                tweenTime = 0;
+            }
+        }
+
+        private void Start()
+        {
+            cauldron.PotionBrewed += DefineColor;
+            cauldron.PotionAccepted += ResetColor;
+            cauldron.PotionDeclined += PotionDeclined;
+        }
+
+        private void OnDestroy()
+        {
+            cauldron.PotionBrewed -= DefineColor;
+            cauldron.PotionAccepted -= ResetColor;
+            cauldron.PotionDeclined -= PotionDeclined;
+            StopAllCoroutines();
```

Comment:
All coroutines on Monobehaviour are stopped automatically, so you don't need this line here.